python/ray/dag/compiled_dag_node.py

```diff
@@ -1875,12 +1874,9 @@ def teardown(self, kill_actors: bool = False):
 
                 if not do_teardown:
                     # Teardown is already being performed.
-                    while True:
-                        with self.in_teardown_lock:
-                            if self._teardown_done:
-                                return
-
-                        time.sleep(0.1)
+                    with self.in_teardown_cond:
+                        self.in_teardown_cond.wait_for(lambda: self._teardown_done)
```

Comment:
[nit] I'm ok with Condition, but since we don't really need a lambda (there's only 1 possible state change: not done -> done) maybe we can just use Event. but if you feel this is good, I'm ok with it.

Reply:
on 2nd thought, maybe even Event is not needed. What if we just use 1 big mutex to lock a single boolean + all teardown logic?

```py
import threading
import time

class TeardownManager:
    def __init__(self):
        self._lock = threading.Lock()
        self._is_done = False

    def teardown(self):
        with self._lock:
            if self._is_done:
                print("Teardown already completed. Returning immediately.")
                return

            # perform the slow process
            print("Performing teardown (slow process)...")
            time.sleep(5)  # Simulate a slow operation
            print("Teardown completed.")

            # mark as done
            self._is_done = True
```

Reply:
ya that makes sense, updated, don't see any reason from before for the extra complexity